### PR TITLE
chore(zephyr): upgrade to Zephyr v3.7.0

### PR DESCRIPTION
**Pull request by gminn**
### Summary

There are several fixes for this app to build with Zephyr 3.7.0 and the latest
Memfault SDK:

- Updated s3 board overlay & board argument to match hardware model v2
- Updated heap pool size to eliminate warning compile message -- they increased it
- Removed enabling of `CONFIG_MEMFAULT_TLS_CERTS_USE_DER` since it is now the
  default
- Updated readme with new args for `wifi connect` shell command
- Enabled `CONFIG_MBEDTLS_AES_ROM_TABLES` explicitly -- this used to be enabled
  by default. It saves ~8kB in RAM space.

With the jump to 3.7.0, the build for the esp32c3 overflowed by 70kb:

```
/Users/gminnehan/.local/zephyr-sdk-0.16.1-rc1/riscv64-zephyr-elf/bin/../lib/gcc/riscv64-zephyr-elf/12.2.0/../../../../riscv64-zephyr-elf/bin/ld.bfd: zephyr/zephyr_pre0.elf section `.dram0.bss' will not fit in region `dram0_0_seg'
/Users/gminnehan/.local/zephyr-sdk-0.16.1-rc1/riscv64-zephyr-elf/bin/../lib/gcc/riscv64-zephyr-elf/12.2.0/../../../../riscv64-zephyr-elf/bin/ld.bfd: DRAM segment data does not fit.
/Users/gminnehan/.local/zephyr-sdk-0.16.1-rc1/riscv64-zephyr-elf/bin/../lib/gcc/riscv64-zephyr-elf/12.2.0/../../../../riscv64-zephyr-elf/bin/ld.bfd: region `dram0_0_seg' overflowed by 71104 bytes
```

After investigating, there appears to be a few culprits, including a drop in the
DRAM segment by 20kB in 3.7.0. To unblock build testing with Memfault for both
the RISC-V and Xtensa arch on Zephyr 3.7.0, the MBEDTLS heap has been shrunk for
the c3 build for now. Users can still test out Memfault shell commands but will
have to post chunks via the chunks debug log. Since this support has been
removed, the default board is now the s3.

### Test Plan

CI, also flashed a esp32s3 locally & uploaded chunks successfully

----
Related: MCU-569

### CODE REVIEW

**Comment by gminn** [comment]
* **#10** <a href="https://app.graphite.dev/github/pr/memfault/zephyr-esp32-example/10?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gminn and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by gminn — COMMENTED** [review]
Need to fix c3 build, somehow we have a 71k dram segment overflow 🤔

**Comment by noahp** [comment]
Oof, 70kB overflow on the esp32c3 :open_mouth: needs investigating.

File: prj.conf [hunk 1]
```diff
@@ -28,8 +28,6 @@ CONFIG_NET_TCP=y
 CONFIG_NET_LOG=y
 CONFIG_INIT_STACKS=y
 
-CONFIG_NET_SHELL=y
```
[the remark this answers is not in the document]
Comment (gminn) [reply prj.conf]: Added back and then set `CONFIG_NET_SHELL=n` in the c3 overlay config

**Pull request merged**